diff --git a/node_modules/react-native-reanimated/android/build.gradle b/node_modules/react-native-reanimated/android/build.gradle
index 6b73a69..ff8a6fc 100644
--- a/node_modules/react-native-reanimated/android/build.gradle
+++ b/node_modules/react-native-reanimated/android/build.gradle
@@ -30,7 +30,7 @@ def resolveReactNativeDirectory() {
     def reactNativePackage = file(
         providers.exec {
             workingDir(rootDir)
-            commandLine("node", "--print", "require.resolve('react-native/package.json')")
+            commandLine(System.getProperty("expo.node.path", "node"), "--print", "require.resolve('react-native/package.json')")
         }.standardOutput.asText.get().trim()
     )
     if (reactNativePackage.exists()) {
@@ -52,7 +52,7 @@ def resolveReactNativeWorkletsDirectory() {
     def reactNativeWorkletsPackage = file(
         providers.exec {
             workingDir(rootDir)
-            commandLine("node", "--print", "require.resolve('react-native-worklets/package.json')")
+            commandLine(System.getProperty("expo.node.path", "node"), "--print", "require.resolve('react-native-worklets/package.json')")
         }.standardOutput.asText.get().trim()
     )
     if (reactNativeWorkletsPackage.exists()) {
@@ -278,7 +278,7 @@ android {
 
 def validateReactNativeVersionResult = providers.exec {
     workingDir(projectDir.path)
-    commandLine("node", "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION.toString())
+    commandLine(System.getProperty("expo.node.path", "node"), "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION.toString())
     ignoreExitValue = true
 }
 
@@ -303,7 +303,7 @@ preBuild.dependsOn(assertNewArchitectureEnabledTask)
 
 def validateWorkletsBuildResult = providers.exec {
     workingDir(projectDir.path)
-    commandLine("node", "./../scripts/validate-worklets-build.js")
+    commandLine(System.getProperty("expo.node.path", "node"), "./../scripts/validate-worklets-build.js")
     ignoreExitValue = true
 }
 
